compile.php

```diff
@@ -28,12 +28,47 @@ function strip_decl($str)
 foreach($refs as $ref)
 {
     $ref = trim($ref);
-    $safedoi = str_replace(')', '_', str_replace('(', '_', str_replace('/', '_', $ref)));
-    $outfile = "tmp/ref_$safedoi.xml";
-    if(!file_exists($outfile))
+    if(strpos($ref, 'doi:') === 0)
     {
-        exec("curl --location --header \"Accept: application/unixref+xml\" \"http://dx.doi.org/$ref\" -o \"$outfile\" " . $redirect, $output, $return_code);
-        $return |= $return_code;
+        $ref = substr($ref, strlen('doi:'));
+        $safedoi = str_replace(')', '_', str_replace('(', '_', str_replace('/', '_', $ref)));
+        $outfile = "tmp/ref_$safedoi.xml";
+        if(!file_exists($outfile))
+        {
+            exec("curl --location --header \"Accept: application/unixref+xml\" \"http://dx.doi.org/$ref\" -o \"$outfile\" " . $redirect, $output, $return_code);
+            $return |= $return_code;
+        }
+    }
+    else if(strpos($ref, 'isbn:') === 0)
+    {
+        $ref = substr($ref, strlen('isbn:'));
+        $isbn = $ref;
+        $outfile = "tmp/ref_$isbn.xml";
+        if(file_exists($outfile))
+            continue;
+            
+        $response = file_get_contents('https://www.googleapis.com/books/v1/volumes?q=isbn:' . $isbn);
+        $results = json_decode($response);
+        if(!isset($results->items) || !count($results->items))
+        {
+            $return |= 1;
+            continue;
+        }
+        $book = $results->items[0];
+        $fp = fopen($outfile, 'w+');
```

Comment:
Thank you so much!
I had forgotten about basex. The reason not to use SimpleXML was to avoid requiring more dependencies (php-xml), but this was probably not right.
BaseX is more elegant and also easier for sure. Though, I still get different output formats for ISBN and DOI references, which means I need to parse two kinds of XML data in my XSLT stylesheet, which is kind of annoying. So I was thinking maybe the treatment should be unified at a higher level, and I should produce an intermediate, home-made format at a lower level. But that's a lot of complications for not so much since I already got this working. I'll probably do what you said nonetheless.